lib/screens/explore_screens/regions_screens/edit_region/interactor/viewmodel/edit_region_bloc.dart

```diff
@@ -38,18 +36,19 @@ class EditRegionBloc extends Bloc<EditRegionEvent, EditRegionState> {
 
   Future<void> _onSaveImageTapped(OnSaveImageTapped event, Emitter<EditRegionState> emit) async {
     emit(state.copyWith(isSaveChangesButtonLoading: true));
-
+    // SAVE image in BUCKET
     final Result<String> urlResult = await SaveImageUseCase()
         .run(SaveImageUseCaseInput(file: state.file!, pathPrefix: PathPrefix.regionImage, creatorId: state.region.id));
+    if (urlResult.isError) {
```

Comment:
This logic needs to be on a mapper. Idk why the previous one is not working, please check how i did it for edit event image. I apply same logic there and it works

Reply:
In the case of the edit event flow, you only have a service(usecase) to save the image and get the url, but this is just one step in edit an event. Once the user reach the end you update the event with that url and the other changes. 

Here it is very different because it is only a screen for one change (image). 
You have in the `on save` 2 services (save in the bucket, update the firebase region).